docs/source/advanced/java-rxjava.mdx

```diff
@@ -0,0 +1,214 @@
+dependencies {
+  // ...
+
+  // For RxJava 2
+  implementation("com.apollographql.apollo3:apollo-rx2-support:$version")
+
+  // For RxJava 3
+  implementation("com.apollographql.apollo3:apollo-rx3-support:$version")
+}
+```
+
+### Queries, Mutations, Subscriptions
+
+Use the `rxSingle` / `rxFlowable` extensions to execute your operations and get RxJava observables:
+
+<MultiCodeBlock>
+
+```java
+// A query
+ApolloQueryCall<MyQuery.Data> queryCall = client.query(new MyQuery());
+Single<ApolloResponse<MyQuery.Data>> queryResponse = rxSingle(queryCall);
+queryResponse.subscribe( /* ... */ );
+
+// A mutation
+ApolloMutationCall<MyMutation.Data> mutationCall = client.mutate(new MyMutation("my-parameter"));
+Single<ApolloResponse<MyMutation.Data>> mutationResponse = rxSingle(mutationCall);
+mutationResponse.subscribe( /* ... */ );
+
+// A subscription
+ApolloSubscriptionCall<MySubscription.Data> subscriptionCall = client.subscribe(new MySubscription());
+Flowable<ApolloResponse<MySubscription.Data>> subscriptionResponse = rxFlowable(subscriptionCall);
+subscriptionResponse.subscribe( /* ... */ );
+```
+
+```kotlin
+// A query
+val queryCall: ApolloQueryCall<MyQuery.Data> = client.query(MyQuery())
+val queryResponse: Single<ApolloResponse<MyQuery.Data>> = queryCall.rxSingle()
+queryResponse.subscribe(/* ... */)
+
+// A mutation
+val mutationCall = client.mutate(MyMutation("my-parameter"))
+val mutationResponse: Single<ApolloResponse<MyMutation.Data>> = mutationCall.rxSingle()
+mutationResponse.subscribe(/* ... */)
+
+// A subscription
+val subscriptionCall = client.subscribe(MySubscription())
+val subscriptionResponse: Flowable<ApolloResponse<MySubscription.Data>> = subscriptionCall.rxFlowable()
+subscriptionResponse.subscribe(/* ... */)
+```
+</MultiCodeBlock>
+
+### ApolloInterceptor
+
+To set an `ApolloInterceptor` on your `ApolloClient`, you can define an `Rx2ApolloInterceptor` or `Rx3ApolloInterceptor`:
+
+<MultiCodeBlock>
+
+```java
+    private static Rx2ApolloInterceptor rx2ApolloInterceptor = new Rx2ApolloInterceptor() {
+        @NotNull
+        @Override
+        public <D extends Operation.Data> Flowable<ApolloResponse<D>> intercept(
+                @NotNull ApolloRequest<D> request,
+                @NotNull Rx2ApolloInterceptorChain chain
+        ) {
+            ApolloRequest<D> modifiedRequest = request.newBuilder() /* ... */.build();
+            return chain.proceed(modifiedRequest);
+        }
+    };
+```
+
+```kotlin
+private val rx2ApolloInterceptor = object : Rx2ApolloInterceptor {
+    override fun <D : Operation.Data> intercept(
+        request: ApolloRequest<D>,
+        chain: Rx2ApolloInterceptorChain
+    ): Flowable<ApolloResponse<D>> {
+        val modifiedRequest = request.newBuilder() /* ... */ .build()
+        return chain.proceed(modifiedRequest)
+    }
+}
+```
+</MultiCodeBlock>
+
+
+Then use the `toApolloInterceptor` extension to convert it to an `ApolloInterceptor` and add it to your `ApolloClient.Builder`:
```

Comment:
Use the fully qualified class name here ? If I'm not mistaken, something like `ToCoroutinesKt. toApolloInterceptor`?

(That doesn't look too good TBH, certainly something to fine tune)

Reply:
Oh yeah I can explicitly show the import and yeah importing the function also works in Java but I agree it's just clearer if we show it here.